File: objectiveai-py/objectiveai/agent/http.py
"""HTTP functions for agent endpoints."""

from __future__ import annotations

from typing import TYPE_CHECKING

if TYPE_CHECKING:
    from objectiveai.agent import GetAgent, ListAgent, UsageAgent
    from objectiveai.client import ObjectiveAI


async def list_agents(client: ObjectiveAI) -> ListAgent:
    """List all agents that have been used."""
    return await client.get_unary("agents")


async def get_agent(client: ObjectiveAI, agent_id: str) -> GetAgent:
    """Retrieve a specific agent by its content-addressed ID."""
    return await client.get_unary(f"agents/{agent_id}")


async def get_agent_usage(client: ObjectiveAI, agent_id: str) -> UsageAgent:
    """Retrieve usage statistics for a specific agent."""
    return await client.get_unary(f"agents/{agent_id}/usage")

File: objectiveai-rs/src/agent/http.rs
//! HTTP client functions for Agent endpoints.

use crate::{HttpClient, HttpError};

/// Lists all Agents that have been used.
pub async fn list_agents(
    client: &HttpClient,
) -> Result<super::response::ListAgent, HttpError> {
    client
        .send_unary(reqwest::Method::GET, "agents", None::<String>)
        .await
}

/// Retrieves a specific Agent by its content-addressed ID.
pub async fn get_agent(
    client: &HttpClient,
    agent_id: &str,
) -> Result<super::response::GetAgent, HttpError> {
    client
        .send_unary(
            reqwest::Method::GET,
            &format!("agents/{}", agent_id),
            None::<String>,
        )
        .await
}

/// Retrieves usage statistics for a specific Agent.
pub async fn get_agent_usage(
    client: &HttpClient,
    agent_id: &str,
) -> Result<super::response::UsageAgent, HttpError> {
    client
        .send_unary(
            reqwest::Method::GET,
            &format!("agents/{}/usage", agent_id),
            None::<String>,
        )
        .await
}
